File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        for (int i = 0; i < targets.length; i++) {
            answer[i] = -1; // -1로 모두 초기화
        }

        Map<Character, Integer> map = new HashMap<>();

        for (String str : keymap) {
            char[] arr = str.toCharArray();
            for (char c : arr) {
                int index = str.indexOf(c) + 1;
                if (map.containsKey(c)) {
                    // 키가 포함된 가장 가까운 위치 값 저장
                    map.put(c, Math.min(map.get(c), index));
                } else {
                    map.put(c, index);
                }
            }
        }

        for (int i = 0; i < targets.length; i++) {
            char[] arr = targets[i].toCharArray();
            int count = 0;
            boolean flag = true;
            for (char c : arr) {
                // 포함되어 있다면 계산
                if (map.containsKey(c)) {
                    count += map.get(c);
                } else {
                    flag = false;
                    break;
                }
            }
            if (flag) answer[i] = count;
        }

        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.kt
import java.lang.IllegalArgumentException
import java.lang.Integer.min
import java.lang.StringBuilder
import java.util.Scanner
import java.util.Stack
import kotlin.math.abs

class Solution {
    fun solution(keymap: Array<String>, targets: Array<String>): IntArray {
        val answer = IntArray(targets.size) { -1 } // -1로 모두 초기화

        val map = mutableMapOf<Char, Int>()

        keymap.forEach { str ->
            val arr = str.toCharArray()
            arr.forEach {
                if (map.containsKey(it)) {
                    //키가 포함된 가장 가까운 위치 값 저장
                    map[it] = min(map[it]!!, arr.indexOf(it) + 1)
                } else {
                    map[it] = arr.indexOf(it) + 1
                }
            }
        }

        for (i in targets.indices) {
            val arr = targets[i].toCharArray()
            var count = 0
            var flag = true
            for (j in arr.indices) {
                //포함되어 있다면 계산
                if (map.containsKey(arr[j])) {
                    count += map[arr[j]]!!
                } else {
                    flag = false
                    break
                }
            }
            if (flag) answer[i] = count
        }
        return answer
    }
}